src/unitxt/metrics.py

```diff
@@ -953,10 +998,91 @@ def aggregation_function(
         return scores_to_resample, aggregation_function
 
     @abstractmethod
-    def compute(self, references: List[Any], prediction: Any, task_data: Dict) -> dict:
+    def compute(
+        self, references: List[Any], prediction: Any, task_data: List[Dict]
+    ) -> dict:
         pass
 
 
+# class Scanner (Metric):
+#     def process_instances(self, instance:Dict[str, Any], instance_metric: InstanceMetric):
+#         if "score" not in instance:
+#             instance["score"] = {"global": {}, "instance": {}}
+#         if "instance" not in instance["score"]:
+#             instance["score"]["instance"] = {}
+#         if "global" not in instance["score"]:
+#             instance["score"]["global"] = {}
+#         self._validate_prediction(instance["prediction"])
+#         self._validate_reference(instance["references"])
+#         instance_score = instance_metric.compute(references=instance["references"], prediction=instance["prediction"], task_data=instance["task_data"] if "task_data" in instance else [{}])
+#         instance["score"]["instance"].update(instance_score)
+#
+
+
+# aggregates (mean, max, or var) numeric scores read from a given field (score_field) found in instance["score"]["instance"]
+# and stores the result aggregated over instances into field named aggregate_score_field
+# (e.g  max_accuracy) in instance["score"]["global"]
+# potentially, the aggregation is carried over subsets of the instances, as induced by filter_grouper
+# the scores in instance["score"]["instance"] are expected to be there following the processing of an
+# InstanceMetric's  compute_instance_scores(stream)
+
+
+class Aggregator(Artifact):
```

Comment:
Is this related to this PR?

Reply:
A start of an attempt to unify the aggregators over the InstanceMetrics